src/cms/preview-templates/Preview.tsx

```diff
@@ -0,0 +1,63 @@
+import React, {
+  FunctionComponent,
+  SetStateAction,
+  Dispatch,
+  useState,
+} from 'react'
+import { Locale, LOCALES } from '../../data/i18n'
+import * as R from 'ramda'
+
+interface PreviewProps<JSONType, ComponentProps> {
+  Component: FunctionComponent<ComponentProps>
+  data: JSONType
+  translator: (locale: Locale) => (data: JSONType) => ComponentProps
+}
+
+export const createPreview = <
+  JSONType extends {} = any,
+  ComponentProps extends {} = any
+>(): FunctionComponent<PreviewProps<JSONType, ComponentProps>> => ({
+  Component,
+  data,
+  translator,
+}) => {
+  const [locale, setLocale] = useState<Locale>('de')
+  const componentProps: ComponentProps = translator(locale)(data)
+  return (
+    <>
+      <LocalePicker currentLocale={locale} setLocale={setLocale} />
+      <Component {...componentProps} />
+    </>
+  )
+}
+
+const LocalePicker: FunctionComponent<{
+  currentLocale: Locale
+  setLocale: Dispatch<SetStateAction<Locale>>
+}> = ({ currentLocale, setLocale }) => (
+  <ul className="nav nav-tabs sticky-top border-0">
+    {R.map(
+      (locale: Locale) => (
+        <LocaleTab
+          locale={locale}
+          key={locale}
+          active={R.equals(currentLocale, locale)}
+          onClick={() => setLocale(locale)}
+        />
+      ),
+      LOCALES,
+    )}
+  </ul>
+)
+
+const LocaleTab: FunctionComponent<{
+  locale: Locale
+  active: boolean
+  onClick: () => void
+}> = ({ locale, active, onClick }) => (
+  <li className="nav-item">
+    <button className={`nav-link ${active ? 'active' : ''}`} onClick={onClick}>
```

Comment:
Just for convenience we could disable the active locale button so there is no pointer for on hovering with the mouse.

```suggestion
    <button className={`nav-link ${active ? 'active' : ''}`} disabled={active} onClick={onClick}>
```